tests/contributing.py

```diff
@@ -0,0 +1,123 @@
+        'Reverse Transform',
+        (
+            'The Transformer can reverse transform the data it produces, going back to the ',
+            'original data type.',
+        ),
+    ),
+    '_validate_composition': (
+        'Composition is Identity',
+        (
+            'Transforming data and reversing it recovers the original data, if composition is ',
+            'identity is specified.',
+        ),
+    ),
+    '_validate_hypertransformer_transformed_data': (
+        'Hypertransformer can transform',
+        'The HyperTransformer is able to use the Transformer and produce float values.',
+    ),
+    '_validate_hypertransformer_reverse_transformed_data': (
+        'Hypertransformer can reverse transform',
+        (
+            'The HyperTransformer is able to reverse the data that it has previously transformed ',
+            'and restore the original data type.',
+        ),
+    ),
+}
+
+
+def get_class(class_name):
+    """Get the specified class.
+
+    Args:
+        class (str):
+            Full name of class to import.
+    """
+    obj = None
+    if isinstance(class_name, str):
+        package, name = class_name.rsplit('.', 1)
+        obj = getattr(importlib.import_module(package), name)
+
+    return obj
+
+
+def validate_transformer_integration(transformer):
+    """Validate the integration tests of a transformer.
+
+    This function runs the automated integration test functions on the Transformer.
+    It will print to console a summary of the integration tests, along with which
+    checks have passed or failed.
+
+    Args:
+        transformer (string or rdt.transformers.BaseTransformer):
+            The transformer to validate.
+    Output:
+        bool:
+            Whether or not the transformer passes all integration checks.
+    """
+    print(f'Validating Integration Tests for transformer {transformer}\n')
+
+    if isinstance(transformer, str):
+        transformer = get_class(transformer)
+
+    steps = []
+    validation_error = None
+    error_trace = None
+
+    try:
+        validate_transformer(transformer, steps=steps)
+    except Exception as error:
+        error_trace = ''.join(traceback.TracebackException.from_exception(error).format())
+
+        for check in CHECK_DETAILS:
+            if check in error_trace:
+                validation_error = str(error)
+
+    if validation_error is None and error_trace is None:
+        print('SUCCESS: The integration tests were successful.\n')
+    elif validation_error:
+        print('ERROR: One or more integration tests were NOT successful.\n')
+    elif error_trace:
+        print('ERROR: Transformer errored out with the following error:\n')
+        print(error_trace)
+
+    result_summaries = []
+    seen_checks = set()
+    failed_step = None if validation_error is None else steps[-1]
+    for i in range(len(steps)):
+        check, details = CHECK_DETAILS[steps[i]]
+        if check in seen_checks:
```

Comment:
cool! so I suppose this skips the `_validate_composition` step if it was not run, right?

Reply:
Yeah `_validate_composition` will not be in the `steps` list if it was not run, so we will not add it to the printed results summary.

This `check in seen_checks` line makes sure that each check only shows up in the table once, since we run some checks multiple times (over multiple dataset generators).

Reply:
Oh, you are right. I misread this line.
If that's the case, maybe we could using a `set` would fix the duplicates problem without having to check here